tsl/src/nodes/decompress_chunk/exec.c

```diff
@@ -1027,9 +1028,11 @@ decompress_get_next_tuple_from_batch(DecompressChunkState *chunk_state,
 		if (is_valid_tuple)
 		{
 			Assert(!TTS_EMPTY(decompressed_slot_projected));
```

Comment:
I think method name `decompress_chunk_perform_select_project` is misleading; it does a filter and projection.
...and the comment inside it about a `1000 rows` looks invalid

Reply:
Do you have a suggestion on how the function should be named?

Could you elaborate on what is wrong with the comment?

Reply:
I think `decompress_chunk_perform_filter_project` makes more sense - as `select` in this context is `project`

The comment suggests that it will only reset the ctx after a ~1000 ;I think thats not true because:
* it does it during every method call
* and the method seem to be invoked for-each-row
  * on one branch it also increments the number of filtered rows by 1

in the current form - whatever caching that ctx could do is thrown away; so its not able to cache anything...but we have a cloudy comment why we are doing that

if you look at the [original changeset](https://github.com/timescale/timescaledb/pull/5421/files#diff-5d87545d5373a40a1ca15cf394c31d0478fe0fa1d0de0c200b170518f0c1a94aR478-R486) which added this comment - the situation was much different

Reply:
> I think decompress_chunk_perform_filter_project makes more sense - as select in this context is project
No, that's the proper term, it's just the relational algebra and SQL conspiracy to confuse us -- `SELECT` is a projection and `WHERE` is a selection :)

Regarding the comment, I also noticed this, because it started showing up in profiles. Anyway, let's keep this for the later, this PR should be a quick backportable fix.

Reply:
sure; that also makes sense; but in that case - the term is `selection` and not `select` ; if you look at explain plans it uses the word `filter`... and that method internally also invokes `InstrCountFiltered1` which  ; I'm just saying that because of the way the method was named I had to read it thru because of the confusion => I think the name of the method did a bad job in describing what it does....maybe: `decompress_chunk_perform_selection_project`  - but no big deal.

> [...] it started showing up in profiles [...]

ok - and how much it worth? I guess it worth the most when there is an udf on a segmentby column.
but I think this comment should be removed...as its not valid anymore

Reply:
>  ok - and how much it worth? 

Not much, 1-2% of the total query CPU time.